src/node_messaging.cc

```diff
@@ -128,12 +175,62 @@ class SerializerDelegate : public ValueSerializer::Delegate {
     env_->isolate()->ThrowException(Exception::Error(message));
   }
 
+  Maybe<bool> WriteHostObject(Isolate* isolate, Local<Object> object) override {
+    if (env_->message_port_constructor_template()->HasInstance(object)) {
+      return WriteMessagePort(Unwrap<MessagePort>(object));
+    }
+
+    env_->ThrowError("Cannot serialize unknown type of host object");
+    return Nothing<bool>();
+  }
+
+  Maybe<uint32_t> GetSharedArrayBufferId(
+      Isolate* isolate, Local<SharedArrayBuffer> shared_array_buffer) override {
+    uint32_t i;
+    for (i = 0; i < seen_shared_array_buffers_.size(); ++i) {
+      if (seen_shared_array_buffers_[i] == shared_array_buffer)
+        return Just(i);
+    }
+
+    SharedArrayBufferMetadataReference reference(
+        SharedArrayBufferMetadata::ForIncomingSharedArrayBuffer(env_,
+                                                          context_,
+                                                          shared_array_buffer));
+    if (!reference) {
+      return Nothing<uint32_t>();
+    }
+    seen_shared_array_buffers_.push_back(shared_array_buffer);
+    msg_->AddSharedArrayBuffer(reference);
+    return Just(i);
```

Comment:
Is the returned ID actually used anywhere in Ayo code or is it just for V8?

Reply:
Only for V8, to match the objects, really… the matching call on the deserializing side is `deserializer.TransferSharedArrayBuffer(i, sab);`